### ISSUE TITLE
Design UI for providing links to valid CSVs for reports

### ISSUE THREAD

**Issue by noahmanger** [18F/openFEC-web-app#1552]
So that users can view the entire contents of a report as a CSV, provide a link to the new CSV files for reports. 

---

We currently provide links to PDF views of reports. We will soon have CSVs of each report, and eventually will have an HTML view ( #1551 ). The task here is to design the means by which people can access each.

The main bit of user data we have on this task comes internal users at the FEC who have requested that the links be inline in the table (rather than in the details panel) since accessing the reports is something that they do many times a day (so it pays to save a click). 

One additional thing to consider is the fact that this download won't require the multi-step process that exporting data from the table does (i.e. the link we provide will be the link to the file itself).

This probably doesn't take a ton of design work, but I wanted to make an issue for us to discuss the different approaches and settle on one before we move to implementation.

cc @jenniferthibault @nickykrause

**Comment by PaulClark2** [comment]
URL Format
http://docquery.fec.gov/csv/XXX/REPID.csv
XXX - last 3 digits of REPID

REPIDs bewteen 1 to 99 are stored in 000 folder

Examples:
http://docquery.fec.gov/csv/000/86.csv

http://docquery.fec.gov/csv/002/1041002.csv
http://docquery.fec.gov/csv/080/256080.csv
http://docquery.fec.gov/csv/105/105.csv

**Comment by noahmanger** [comment]
Ok, I have a few different concepts to share. 

## Option 1
This is to basically include a link to the CSV inline with the document description (which currently links to a PDF, but could link to the HTML page next). This could look like this:
![image](https://cloud.githubusercontent.com/assets/1696495/18934166/1b346622-858d-11e6-9071-0c04b5d799e6.png)

Or we could style the CSV links as `.button--alt` and right-align like so:
![image](https://cloud.githubusercontent.com/assets/1696495/18934178/265bb992-858d-11e6-9d23-a72e395c7187.png)

Because there's additional download options (PDF and potentially .FEC), the details panel could be revised to present those options like this:

![image](https://cloud.githubusercontent.com/assets/1696495/18934234/886d939e-858d-11e6-94e4-5878f56615dc.png)

This presents a "View report" button which would go to the HTML page and a "Download" button like the Calendar "Download" button which includes the various options. 

Pro's to Option 2:
- Keeps the table simple while still including a link to the most generally useful download file directly in the table

Cons:
- Perhaps annoying to power users to not have all the options directly in the table


## Option 2
This would make a new column at the end that includes links to PDF, CSV and .FEC files like so:
![image](https://cloud.githubusercontent.com/assets/1696495/18934187/3c308a5e-858d-11e6-9f5a-44f6a9ef00df.png)

Pros:
- Exposes more options directly in the table
- Is most similar to what they currently do, thus making it familiar to experienced users at FEC and outside

Cons:
- Clutters the table
- Perhaps exposes too many options up front

---

For further context, I want to think through what we know and don't know about this interaction. 

What we (believe we) know:
- .FEC files are primarily used by users inside the FEC and are basically useless outside
- Users inside the FEC frequently download these files as a key function of their job, and do so over and over
- .CSV files are the best way for outside users to access raw data

What we (believe we) don't know:
- I'm not sure _how_ common the task of downloading .CSV files is for people outside the FEC
- We don't know if we'll be able to get to a scenario where we don't need to provide .FEC files, as the software used internally relies on this file format (but might be something we can get around in the future)

---

Anyway, lots of notes on what's ultimately a pretty small interaction. Curious what folks think.

cc @jenniferthibault @nickykrause @onezerojeremy

**Comment by jenniferthibault** [comment]
@noahmanger thanks for getting these options together. To me it feel like given already tight screens, if we can avoid adding an extra column, we should.

Exposing more options directly in the table (a "pro" from option 2), but in a way that clutters less and combines some of your Option 1 ideas:

![screen shot 2016-09-29 at 12 50 59 pm](https://cloud.githubusercontent.com/assets/11636908/18963518/51011d84-8643-11e6-8c78-d707d3bf0b4c.png)


This keeps all the download options up front. It still takes a click, but might not be perceived as going any deeper into layers of the page, like opening the panel does.

**Comment by noahmanger** [comment]
Oooh I like it. Nice application of the "add event" interaction from the calendar. 
![image](https://cloud.githubusercontent.com/assets/1696495/18963816/4b341de2-862b-11e6-95d7-ed5ebc0a0dcf.png)

It feels like it strikes a good balance of exposing interactions up front while not cluttering the table.

Interested in @PaulClark2 and @jwchumley 's thoughts, given that this is a very "power user" interaction.

**Comment by onezerojeremy** [comment]
Oh nice! If pro users download a lot of files at once, this is pretty nice as it's only two clicks per download. 

It might be nice to figure out a way to make the file-type selection stick between downloads to save a click, (but I'd mark it as a nice to have or future enhancement*). 


* also if it's a pro-feature one could imagine keyboard navigation of the results list, much the way gmail allows. definitely in the future enhancement category.

**Comment by LindsayYoung** [comment]
For filings, we should have the csv, pdf and fec exposed like in option 2, if possible. RAD has to go and download these every day- I don't want to give them more clicks than we have to.

**Comment by noahmanger** [comment]
@LindsayYoung just want to make sure I'm understanding you: you're suggesting that on the /filings page (but not the reports pages I posted above) we use Option 2? 

I'm not opposed to trying that, I'm a little wary of having different patterns in different places, though maybe it's not a problem.

**Comment by LindsayYoung** [comment]
Yea, I don't have an opinion about the display on the reports page. I generally like consistency, so I would lean towards exposing them everywhere.

But what I really care about is, if we can get single click downloads on the .fec files on the filings page that will save RAD time and redundancy- they might be the most avid users of that page so I think their needs there are important here. Their most frequent task is to download the filings so they can inspect the filings in their tools.

**Comment by onezerojeremy** [comment]
We have a few meetings with RAD users on Wednesday; I'd be happy to demo a couple options! 

Right now we're forcing a choice, but what if we set a default? Maybe by breaking the button into two parts, one that prompts the choice, the other which downloads the selected?

**Comment by noahmanger** [comment]
+1 to running something by folks.

I like where you're going with that @onezerojeremy . Here's a similar (identical?) pattern from Bootstrap: http://getbootstrap.com/components/#btn-dropdowns-split

**Comment by onezerojeremy** [comment]
Oh my yes. that's identical to what i was thinking!

**Comment by noahmanger** [comment]
Circling back on this: since the API support is just about ready and we haven't gotten a clear signal from RAD, I think we should implement a first version of this using existing components, so what @jenniferthibault mocked up above. That's low opportunity cost and then we can refine once we do usability testing with the live interaction. How does that sound @onezerojeremy ?

**Comment by onezerojeremy** [comment]
I nudged again for feedback today, and sent the mocked up component @jenniferthibault designed, [this one](https://cloud.githubusercontent.com/assets/11636908/18963518/51011d84-8643-11e6-8c78-d707d3bf0b4c.png), so perhaps let's give it another 24 hours or so? (busy time of year for RAD)

From my perspective, given what we heard from the folks in RAD we talked to (both about the frequency of this action, and the need for it to be as lightning fast as possible, and their mock-up
![20161005_164240](https://cloud.githubusercontent.com/assets/474916/19282561/dce426ca-8fbc-11e6-9e2d-29d3096da0bc.jpg), I think Jen's design gets us close. For reference, I think this is how they do it now:

![fec-download](https://cloud.githubusercontent.com/assets/474916/19283052/8722131c-8fbe-11e6-9724-47ae2a170c51.gif)
so it's 
- open bookmarked viewer (really it's probably already open, with filings selected)
-  ctl-v the committee ID and hit enter. 
- click on committee name, 
- click on filing
- click on ascii in pop-up (which they know to be .fec)

So it's Paste, hit enter, + 3 clicks. 

If they bookmark [all filings](https://beta.fec.gov/data/filings/)
- open bookmarked filings search
-  ctl-v the committee ID and hit enter (or click but probably enter)
-  click filing download type selector
- click .fec from selector

So that's one click less! That's an improvement, and probably it's better than that because you won't have to wait on the pop-up to load, plus it's closer ([fitt's law](https://en.wikipedia.org/wiki/Fitts%27s_law) ftw)

**Comment by onezerojeremy** [comment]
Just heard back from one of the very kind analysts we got a chance to talk to, and they thought that this approach seemed good. 

They had one question, which I agreed was a good one: will there be an amendment indicator?

**Comment by LindsayYoung** [comment]
there isn't a way to see if a document has been amended using the current table, we are planning to switch off the warehouse table and go to a better filings view, where that should be possible.

**Comment by noahmanger** [comment]
Talked to Lauren about this yesterday as well and she supports the button approach! The amendment indicator is coming. Closing.

**Comment by noahmanger** [comment]
Oh, she also raised a good point that it was helpful to have this close to the committee name and document description, rather than in the far right column, as it's easier to see what it's associated with. So just extra points for this approach.

**Comment by onezerojeremy** [comment]
fantastic!
